extensions/amp-vr-image/0.1/amp-vr-image.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {addParamToUrl, assertHttpsUrl} from '../../../src/url';
+import {dev} from '../../../src/log';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {loadPromise} from '../../../src/event-helper';
+
+/** @const */
+const TAG = 'amp-vr-image';
+
+class AmpVrImage extends AMP.BaseElement {
```

Comment:
I thought about that. But here's my thought: there's really only one way to show an image. Eventually we could host this viewer on our CDN or even compile it all inline. That being said, we can always start with a more specific name and "promote" it to a more generic one later. WDYT?